onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/VehicleOverlay.java

```diff
@@ -182,21 +188,11 @@ public synchronized void clear() {
      * Cache the core black template Bitmaps used for vehicle icons
      */
     private static final void loadIcons() {
-        // Initialize variables used for all marker icons
-        Resources r = Application.get().getResources();
-
-        // Black vehicle icons
-        if (vehicle_icons[0] == null) {
-            vehicle_icons[0] = createVehicleIcon(NORTH);
-            vehicle_icons[1] = createVehicleIcon(NORTH_EAST);
-            vehicle_icons[2] = createVehicleIcon(EAST);
-            vehicle_icons[3] = createVehicleIcon(SOUTH_EAST);
-            vehicle_icons[4] = createVehicleIcon(SOUTH);
-            vehicle_icons[5] = createVehicleIcon(SOUTH_WEST);
-            vehicle_icons[6] = createVehicleIcon(WEST);
-            vehicle_icons[7] = createVehicleIcon(NORTH_WEST);
-            vehicle_icons[8] = createVehicleIcon(NO_DIRECTION);
-        }
+        vehicle_type_icons.put(ObaRoute.TYPE_BUS, createBusIcons());
```

Comment:
IIRC because `vehicle_type_icons` is static, it can end up being initialized multiple times, each time `VehicleOverlay` is instantiated.  I had the check for `vehicle_icons[0] == null` in the previous code to make sure that we didn't pay this initialization penalty more than once.  This is even more important now that we're loading more vehicle icon types.  You can add a similar IF block around the initialization of `vehicle_type_icons`, something like `if (!vehicle_type_icons.contains(ObaRoute.TYPE_BUS)) {`, to do this same type of check.